File: app/src/main/java/com/minwook/mytriphistory/java/ui/fragment/MapFragment.java
package com.minwook.mytriphistory.ui.fragment;


import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.android.gms.maps.CameraUpdateFactory;
import com.google.android.gms.maps.GoogleMap;
import com.google.android.gms.maps.MapView;
import com.google.android.gms.maps.OnMapReadyCallback;
import com.google.android.gms.maps.SupportMapFragment;
import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.MarkerOptions;
import com.minwook.mytriphistory.R;
import com.minwook.mytriphistory.present.MapPresent;
import com.minwook.mytriphistory.present.MapPresentImpl;

/**
 * A simple {@link Fragment} subclass.
 * Use the {@link MapFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
public class MapFragment extends Fragment implements OnMapReadyCallback, MapPresent.View {
    private GoogleMap mMap;
    private MapPresentImpl mapPresent;
    private String TAG = MapFragment.class.getName();

    private MapView mapView;

    public MapFragment() {
        // Required empty public constructor
    }

    public static MapFragment newInstance() {
        MapFragment fragment = new MapFragment();
        /*Bundle args = new Bundle();
        args.putString(ARG_PARAM1, param1);
        args.putString(ARG_PARAM2, param2);
        fragment.setArguments(args);*/
        return fragment;
    }

    @Override
    public void onActivityCreated(Bundle bundle) {
        super.onActivityCreated(bundle);
        //getMapAsync(this);
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        Log.d(TAG, "onCreate start");
        super.onCreate(savedInstanceState);
        mapPresent = new MapPresentImpl(getContext(), this);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        Log.d(TAG, "onCreateView start");
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_map, container, false);
        mapView = view.findViewById(R.id.mapView);

        mapView.onCreate(savedInstanceState);
        mapView.onResume();

        mapView.getMapAsync(this);
        return view;
    }

    @Override
    public void onMapReady(GoogleMap googleMap) {
        Log.d(TAG, "loadLocations start");
        mMap = googleMap;

        /*
        // Add a marker in Sydney and move the camera
        LatLng sydney = new LatLng(-34, 151);
        mMap.addMarker(new MarkerOptions().position(sydney).title("Marker in Sydney"));
        mMap.moveCamera(CameraUpdateFactory.newLatLng(sydney));
        */

        mapPresent.loadLocations();


        //37.551578, 126.989842 서울 중심
    }

    @Override
    public void addMarker(double latitude, double longtitude, String title) {
        Log.d(TAG, "addMarker start");
        LatLng location = new LatLng(latitude, longtitude);
        mMap.addMarker(new MarkerOptions().position(location).title(title));
    }

    @Override
    public void setCenterMap(double latitude, double longtitude) {
        Log.d(TAG, "setCenterMap start");
        LatLng center = new LatLng(latitude, longtitude);
        mMap.moveCamera(CameraUpdateFactory.newLatLng(center));
        mMap.animateCamera(CameraUpdateFactory.zoomTo(11));
    }
}

File: app/src/main/java/com/minwook/mytriphistory/kotlin/ui/fragment/MapFragment.kt
package com.minwook.mytriphistory.kotlin.ui.fragment

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.google.android.gms.maps.CameraUpdateFactory
import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.OnMapReadyCallback
import com.google.android.gms.maps.model.LatLng
import com.google.android.gms.maps.model.MarkerOptions
import com.minwook.mytriphistory.R
import com.minwook.mytriphistory.kotlin.present.MapPresent
import com.minwook.mytriphistory.kotlin.present.MapPresentImpl
import kotlinx.android.synthetic.main.fragment_map.*

class MapFragment: Fragment(), OnMapReadyCallback, MapPresent.View  {
    private lateinit var mMap: GoogleMap
    private lateinit var mapPresent: MapPresentImpl

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_map, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        mapView.onCreate(savedInstanceState)
        mapView.onResume()
        mapView.getMapAsync(this)

        mapPresent = MapPresentImpl(this)
    }


    override fun onMapReady(googleMap: GoogleMap?) {
        googleMap?.let{
            mMap = it
            mapPresent.loadLocations()
        }
    }

    override fun addMarker(latitude: Double, longtitude: Double, title: String) {
        val location = LatLng(latitude, longtitude)
        mMap.addMarker(MarkerOptions().position(location).title(title))

    }

    override fun setCenterMap(latitude: Double, longtitude: Double) {
        val center = LatLng(latitude, longtitude)
        mMap.moveCamera(CameraUpdateFactory.newLatLng(center))
        mMap.animateCamera(CameraUpdateFactory.zoomTo(11f))
    }
}